subsys/bluetooth/host/smp.c

```diff
@@ -2778,9 +2779,9 @@ bool bt_smp_request_ltk(struct bt_conn *conn, uint64_t rand, uint16_t ediv, uint
 		/* Notify higher level that security failed if security was
 		 * initiated by slave.
 		 */
-		bt_conn_security_changed(smp->chan.chan.conn,
+		bt_l2cap_encrypt_change(conn, BT_HCI_ERR_PIN_OR_KEY_MISSING);
+		bt_conn_security_changed(conn,
```

Comment:
Should bt_conn_security_changed be calling bt_l2cap_encrypt_change? We might as well update the names to be consistent e.g. bt_l2cap_security_changed.

Reply:
Good idea.